python/pyarrow/tests/test_parquet.py

```diff
@@ -29,21 +29,19 @@
 from pyarrow.compat import guid, u, BytesIO, unichar
 from pyarrow.tests import util
 from pyarrow.filesystem import LocalFileSystem
-import pyarrow as pa
 from .pandas_examples import dataframe_with_arrays, dataframe_with_lists
 
 import numpy as np
 import pandas as pd
-
+import pyarrow as pa
+import pyarrow.parquet as pq
```

Comment:
Uh, this test file should be skipped if parquet support wasn't built, instead of raising.

Reply:
Sure, we can use `pytest.importorskip` for that

Reply:
Be careful with `importorskip` because it can silently ignore errors due to linking issues  library loading. These should raise

Reply:
BTW the issue with `importorskip` is exactly why the imports are inside the tests

Reply:
Thanks for the warning! Do we really want to skip if parquet is not available? It wasn't skipped previously.

Reply:
Hmm, I think it was skipped. The tests wouldn't fail if parquet wasn't built.

Reply:
Don't the tests fail in both cases?

Reply:
Unless something changed recently, the tests in this file were only run if either `--parquet` was passed or `import pyarrow.parquet as pq` runs successfully

see

https://github.com/apache/arrow/blob/master/python/pyarrow/tests/conftest.py#L39

Reply:
Fall back to import `pq` in each test function, I'll submit another PR to handle optional dependencies without redundant imports after 0.10.0 release.